source/features/forked-to.tsx

```diff
@@ -30,19 +30,11 @@ const updateCache = cache.function(async (): Promise<string[] | undefined> => {
 	staleWhileRevalidate: 5
 });
 
-function createLink(baseRepo: string): string {
-	if (pageDetect.isRepoRoot() || !(pageDetect.isSingleFile() || pageDetect.isRepoTree())) {
-		return '/' + baseRepo;
-	}
-
-	const [user, repository] = baseRepo.split('/');
-	const url = new GitHubURL(location.href).assign({
-		user,
-		repository,
-		branch: 'HEAD'
-	});
-
-	return url.pathname;
+function mapRepositoryInfoToGitHubURL(repository: Partial<RepositoryInfo>): Partial<GitHubURL> {
+	return {
+		user: repository.owner,
+		repository: repository.name
+	};
```

Comment:
This could be avoided by adding a `Repository` **setter** to GitHubURL.

Reply:
I tried but could not figure it out